zjit/src/hir.rs

```diff
@@ -402,11 +403,18 @@ impl PtrPrintMap {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Copy)]
 pub enum SideExitReason {
     UnknownNewarraySend(vm_opt_newarray_send_type),
     UnknownCallType,
     UnknownOpcode(u32),
+    FixnumAddOverflow,
+    FixnumSubOverflow,
+    FixnumMultOverflow,
+    GuardTypeFixnum,
```

Comment:
Let's just do one `GuardType(Type)` since it's neatly printable and `GuardBitEquals(Value)`